### PR TITLE
Remove stale istio-egress

### PR DESCRIPTION
**Pull request by rshriram**
**What this PR does / why we need it**:
We no longer use istio-egress. It is not documented either. Has been superseded by egress rules.
Also fixed updateVersion.sh to update sidecar version in bookinfo.sidecar.yaml files in Consul/Eureka.
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: samples/bookinfo/consul/bookinfo.sidecars.yaml [hunk 1]
```diff
@@ -16,7 +18,7 @@
 version: '2'
 services:
   details-v1-init:
-    image: istio/proxy_init:0.2.9
+    image: gcr.io/istio-testing/proxy_init:e3f2b2252f76ba579e658764735cd0de10597557
```
Comment (ijsnellf) [inline samples/bookinfo/consul/bookinfo.sidecars.yaml]: is this meant to be a sha?

**Comment by andraxylia** [comment]
Will this task still work?
https://istio.io/docs/tasks/traffic-management/egress.html

**Comment by rshriram** [comment]
yep

**Pull request merged**